File: Task2/apz-pzpi-21-11-milennyi-ivan-task2/src/view_models/feeding_log_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct FeedingLogVM{
    id: u64,
    timestamp: u64,
    amount: u64,
    shepherd_name: Option<String>,
    shepherd_surname: Option<String>,
    sheep_id: u64
}

File: Task4/apz-pzpi-21-11-milennyi-ivan-task4/src/viewModels/FeedingLogVM.ts
import { timestampToDate } from "../utils/helpers";

export class FeedingLogVM {
    id: number;
    date: string;
    amount: number;
    shepherdName: string | null;
    shepherdSurname: string | null;
    sheepId: number;

    constructor(id: number, timestamp: number, amount: number, sheepId: number, shepherdName: string | null, shepherdSurname: string | null) {
        this.id = id;
        this.amount = amount / 1000;
        this.shepherdName = shepherdName;
        this.shepherdSurname = shepherdSurname;
        this.sheepId = sheepId;
        this.date = timestampToDate(timestamp);
    }
}
